knowledge_graphs/ai_script_analyzer.py

```
@@ -0,0 +1,550 @@
+                - imports: List of imported modules
+                - functions: List of function definitions with metadata
+                - classes: List of class definitions with metadata
+                - external_calls: List of detected external API/library calls
+                - file_operations: List of file operation calls
+                - assertions: List of assertions and assumptions made
+                - complexity_score: Estimated complexity score
+                - suspicious_patterns: List of potentially problematic patterns
+        """
+        try:
+            # Read the script content
+            with open(script_path, 'r', encoding='utf-8') as f:
```

Ruff (0.12.2)

Unnecessary mode argument

Remove mode argument

(UP015)